packages/code-editor/src/Markdown/utils.ts

```diff
@@ -0,0 +1,138 @@
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+/**
+ * TODO: Upgrade build to esnext and use `replaceAll`
+ */
+export const regReplaceAll = (
+  content: string,
+  pattern: string,
+  replacementFunc: any
+) => {
+  const target = new RegExp(pattern, 'gi')
+  return content.replace(target, replacementFunc)
+}
+/**
+ * Adds <mark> tags around text if search pattern is detected
+ * @param content - the main content to render
+ * @param searchPattern - the search pattern to consider
+ * @returns - a 'marked' string to be rendered by markdown component
+ */
+export const addMarkTags = (content: string, searchPattern: string) => {
+  let markedContent
+  try {
+    const replacement = (match: string) => `<mark>${match}</mark>`
+    markedContent = regReplaceAll(content, searchPattern, replacement)
+  } catch (e) {
+    markedContent = content
+  }
+  return markedContent
+}
+
+/**
+ * Removes <mark><mark/> tags from markdown text. The mark tag is removed from code styled
+ * text as the code renderer cannot differentiate <mark><mark/> from the code text.
+ * @param markedText - the markdown text input that contains <mark /> tags
+ */
+const removeMarkTags = (markedText: string) => {
+  return markedText.replace(/<mark>/g, '').replace(/<\/mark>/g, '')
+}
+
+/**
+ * Checks the input for code block decorators, the programming language used (```json), and returns a language tag
+ * that is later used to inform syntax highlighting rules
+ * @param content - the code blob to render
+ * @returns - code blob string with code language tag
+ */
+const addCodeLanguageTags = (content: string) => {
+  let languageTaggedContent
+  try {
+    const replacement = () => '```\n<json/>'
+    const searchPattern = '```json'
+    languageTaggedContent = regReplaceAll(content, searchPattern, replacement)
+  } catch (e) {
+    languageTaggedContent = content
+  }
+  return languageTaggedContent
+}
+
+/**
+ * Removes the code language tag from the code blob text before text presentation to screen
+ * @param content - code blob text with language tag
+ * @returns - code blob text without language tag
+ */
+const removeCodeLanguageTags = (content: string) => {
+  let untaggedContent
+  try {
+    const replacement = () => ''
+    const searchPattern = '<json/>'
+    untaggedContent = regReplaceAll(content, searchPattern, replacement)
+  } catch (e) {
+    untaggedContent = content
+  }
+  return untaggedContent
+}
+
+/**
+ * Extracts the syntax highlighting language, if specified
+ * @param content - language tagged code blob
+ * @returns
+ */
+const getCodeLanguageFromTaggedText = (content: string): string => {
+  return content.match(/<json\/>/g) ? 'json' : 'markup'
```

Comment:
I think we'll need to have a conversation before I can review this `json` code further.